### PR TITLE
Replace old Release docs in favor of a cheat sheet

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

The release docs had all the necessary steps but not in the right order (and
sometimes with a lot of extra context). This commit replaces it with a step by
step cheat sheet that can be followed to perform a Triggers release.

Based on https://github.com/tektoncd/pipeline/blob/master/tekton/release-cheat-sheet.md

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
Adds a step by step checklist to follow for releasing Triggers

```

/kind documentation

### CODE REVIEW

File: tekton/README.md [hunk 1]
```diff
@@ -17,144 +17,13 @@ The Pipelines and Tasks in this folder are used for:
 
 To start from scratch and use these Pipelines and Tasks:
 
-1. [Install Tekton v0.3.1](https://github.com/tektoncd/pipeline/blob/master/tekton/README.md#install-tekton)
+1. [Install Tekton](https://github.com/tektoncd/pipeline/blob/master/tekton/README.md#install-tekton)
 1. [Setup the Tasks and Pipelines](https://github.com/tektoncd/pipeline/blob/master/tekton/README.md#setup)
```
Comment (savitaashture) [inline tekton/README.md:21]: @dibyom While reviewing PR found that link is not properly configured to point to exact path in the original code
as this is part of README can you please handle this also as part of this PR itself 
```suggestion
1. [Setup the Tasks and Pipelines](https://github.com/tektoncd/pipeline/blob/master/tekton/README.md#install-tasks-and-pipelines)
```

Comment (dibyom) [reply tekton/README.md]: done!

**Pull request merged**